src/bin/write_value_labels.c

```diff
@@ -0,0 +1,94 @@
+#include "../readstat.h"
+#include <stdio.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <errno.h>
+
+#include "../spss/readstat_sav_date.h"
+#include "../stata/readstat_dta_days.h"
+
+#include "format.h"
+#include "extract_label_set.h"
+#include "extract_metadata.h"
+#include "write_value_labels.h"
+
+void add_val_labels(struct context *ctx, readstat_variable_t *variable, const char *val_labels) {
+    if (!val_labels) {
+        return;
+    } else {
+        fprintf(stdout, "extracting value labels for %s\n", val_labels);
+    }
+    int sav_date = variable->format && variable->format[0] && (strcmp(variable->format, "EDATE40") == 0) && variable->type == READSTAT_TYPE_DOUBLE;
+    int dta_date = variable->format && variable->format[0] && (strcmp(variable->format, "%td") == 0) && variable->type == READSTAT_TYPE_INT32;
+    readstat_label_set_t * label_set = get_label_set(val_labels, ctx, 0);
+    if (!label_set) {
+        fprintf(stderr, "Could not find label set %s!\n", val_labels);
+        exit(EXIT_FAILURE);
+    }
+    fprintf(ctx->fp, ", \"categories\": [");
+    for (int i=0; i<label_set->value_labels_count; i++) {
+        readstat_value_label_t* value_label = &label_set->value_labels[i];
+        if (i>0) {
+            fprintf(ctx->fp, ", ");
+        }
+        if (sav_date) {
+            char* lbl = quote_and_escape(value_label->label);
+            char buf[255];
+            char *s = readstat_sav_date_string(value_label->double_key, buf, sizeof(buf)-1);
+            if (!s) {
+                fprintf(stderr, "%s:%d could not parse double value %lf to date\n", __FILE__, __LINE__, value_label->double_key);
+                exit(EXIT_FAILURE);
+            }
+            fprintf(ctx->fp, "{ \"code\": \"%s\", \"label\": %s} ", s, lbl);
+            free(lbl);
+        } else if (dta_date) {
+            char* lbl = quote_and_escape(value_label->label);
+            char buf[255];
+            int k = value_label->int32_key;
+            char tag = 0;
+            if (k >= 2147483622) {
```

Comment:
I think this number is the same as [DTA_113_MISSING_INT32_A](https://github.com/WizardMac/ReadStat/blob/master/src/stata/readstat_dta.h#L143), defined in [src/stata/readstat_dta.h](https://github.com/WizardMac/ReadStat/blob/master/src/stata/readstat_dta.h).